DeflateWriter.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace samdark\sitemap;
+
+/**
+ * Flushes buffer into file with incremental deflating data, available in php 7.0+
+ */
+class DeflateWriter
+    implements WriterInterface
+{
+    /**
+     * @var resource for target file
+     */
+    private $file;
+
+    /**
+     * @var resource for writable incremental deflate context
+     */
+    private $deflateContext;
+
+    public function __construct($filename)
+    {
+        $this->file = fopen($filename, 'a');
```

Comment:
Worth adding `b` in order for it to be binary-safe.

Reply:
You are of course correct. I thought about it, but did not see 'b' in the list of fopen flags and so thought it was not necessary anymore.